### ISSUE TITLE
Panel not showing all icons

### ISSUE THREAD

**Issue by q-pa** [lumina-desktop/lumina#649]
Fresh install of 18.12-RELEASE. 
The Thunderbird icon is missing from the panel (only showing some default icon). In the settings menu it shows correctly, though.
See attached two screenshots.
![screenshot-panel](https://user-images.githubusercontent.com/3322553/51636342-50396580-1f59-11e9-825d-d715fe12da59.png)

![screenshot-panel-settings](https://user-images.githubusercontent.com/3322553/51636349-5596b000-1f59-11e9-86c7-841e08d438d6.png)

**Comment by beanpole135** [comment]
I just fixed this a moment ago: https://github.com/lumina-desktop/lumina/commit/9ca1d49ef773adae2a26758a61c9db3b029d93b1

**Comment by q-pa** [comment]
![screenshot-panel2](https://user-images.githubusercontent.com/3322553/52005649-81221900-24ca-11e9-9f21-c2510c60b4fb.png)
Update U2 did not resolve this issue. Now, the icon does show the Gnome footprint, though...? Could you please re-open this issue? Thanks!

**Comment by beanpole135** [comment]
Reopening for review again.

**Comment by q-pa** [comment]
On my old TrueOS installation where the panel is showing a thunderbird icon, the only icon I could find is located in the directory: /usr/local/share/icons/pixmaps/apps

**Comment by q-pa** [comment]
FYI, I created this directory and copied the .png file plus I copied two different icon sizes to the hicolor directory... still not working.

**Comment by q-pa** [comment]
Since today's update U6 the icon for AppCafé is another default icon.

**Comment by beanpole135** [comment]
ok, now this is fixed with commit https://github.com/lumina-desktop/lumina/commit/5747f7fe96c2419f8837bafd7122ba2a1ad04497

It was a combination of two issues:
1. The thunderbird XDG shortcut was partially-invalid because it was specifying the file extension (which you are not supposed to do with icons from the global theme). I added a detection/skip of any file extension on a theme icon and that works much better now.
2. The new LFileInfo class that the quicklaunch widgets uses did not have the backwards compatibility for the old pixmaps directory - which was preventing the appcafe icon from getting loaded on those buttons. I enhanced that class to support that old dir again, as well as added the initial file->icon conversion routine within that class itself (since that class is where Lumina is trying to put the handling for all the special kinds of files to make it easier to use across all sorts of different projects).

**Comment by q-pa** [comment]
Updated to U7, Thunderbird icon is showing now but still no AppCafé icon...

**Comment by utrenkner** [comment]
I confirm, this problem persists in 1.5.0. I just upgraded from the current FreeBSD port (1.4.1), yesterday, and several app icons in the panel have been replaced with a default file icon.

![Screenshot-2019-05-28-14-41-31](https://user-images.githubusercontent.com/8254715/58478941-7c5b3200-8157-11e9-8b0c-d465cbb251b1.png)

 In the settings they are still there:
![Screenshot-2019-05-28-14-43-12](https://user-images.githubusercontent.com/8254715/58478947-82511300-8157-11e9-9bf9-28a970c37a30.png)

Lumina was built from the 1.5.0 release sources on FreeBSD 12.0-RELEASE-p4.
